sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1225,7 +1225,7 @@ inline std::string GetConnectionKey(std::string const& host, CurlTransportOption
 {
   std::string key(host);
   key.append(!options.CAInfo.empty() ? options.CAInfo : "0");
-  key.append(!options.Proxy.empty() ? options.Proxy : "0");
+  key.append(options.Proxy ? (options.Proxy->empty() ? "NoProxy" : options.Proxy.Value()) : "0");
```

Comment:
You probably want to check `options.Proxy.HasValue()` before going to `->` since that will misbehave if it has no value (IIRC `Nullable::operator->` assumes that the nullable has a value). Alternatively, you could update `Nullable::operator->` to simply call `Nullable::Value()` and avoid the issue entirely.

Reply:
that is the first condition `options.Proxy ?`  
Usually I don't like nested ternaries, but I just wanted to keep it as one line.

`options.Proxy ?` is the same as `options.Proxy.HasValue()`.